### PR TITLE
AEA-1116 Message class and protocol generator improved for memory saving

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

Message class and protocol generator improved for memory saving

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by DavidMinarsch — COMMENTED** [review]
Missing:
- wrong protoc used
- memory savings statistics

File: benchmark/checks/check_messages_memory_usage.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def run(messages_amount: int):
 
 
 @click.command()
-@click.option("--messages", default=10 ** 6, help="Amount of messages.")
+@click.option("--messages", default=10 ** 5, help="Amount of messages.")
```
Comment (DavidMinarsch) [inline benchmark/checks/check_messages_memory_usage.py:69]: please revert so we keep comparability

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by lrahmani — APPROVED** [review]
LGTM

**Pull request merged**